homeassistant/components/homewizard_climate/climate.py

```diff
@@ -0,0 +1,238 @@
+"""Climate platform for homewizard_climate."""
+import logging
+import time
+
+from homewizard_climate_websocket.model.climate_device_state import (
+    HomeWizardClimateDeviceState,
+)
+from homewizard_climate_websocket.ws.hw_websocket import HomeWizardClimateWebSocket
+
+from homeassistant.components.climate import (
+    FAN_HIGH,
+    FAN_LOW,
+    FAN_MEDIUM,
+    FAN_OFF,
+    FAN_ON,
+    SWING_HORIZONTAL,
+    SWING_OFF,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Create entries for each device in Homewizard cloud."""
+    websockets = hass.data[DOMAIN][entry.entry_id]["websockets"]
+    entities = [HomeWizardClimateEntity(ws, hass) for ws in websockets]
+    async_add_entities(entities)
+
+
+class HomeWizardClimateEntity(ClimateEntity):
+    """Climate entity for a given device in Homewizard cloud."""
+
+    def __init__(
+        self, device_web_socket: HomeWizardClimateWebSocket, hass: HomeAssistant
+    ) -> None:
+        """Initialize the device and identifiers."""
+        self._device_web_socket = device_web_socket
+        self._device_web_socket.set_on_state_change(self.on_device_state_change)
+        self._hass = hass
+        self._logger = logging.getLogger(
+            f"{__name__}.{self._device_web_socket.device.identifier}"
+        )
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device specific attributes."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._device_web_socket.device.identifier)},
+            name=self.name,
+        )
```

Comment:
Yeah, I checked the `/auth/devices` API response when I added this to see if any additional information is returned, unfortunately that wasn't the case. The response is limited to [these fields](https://github.com/mepla/homewizard-climate-websocket/blob/main/homewizard_climate_websocket/model/climate_device.py#L17-L22). There might be a different API endpoint that gives more info about the devices, lmk if you it. Otherwise I'd rather not add any hardcoded values here.